CONTRIBUTING_TO_OPENSHIFT_PLUGINS.md

```diff
@@ -0,0 +1,280 @@
+* [OpenShift Pipeline Plugin (only in maintenance at this point)](https://github.com/openshift/jenkins-plugin)
+
+Per the process for [hosting plugins](https://wiki.jenkins.io/display/JENKINS/Hosting+Plugins) specified by the open source Jenkins
+project, there are associate wikis in the Jenkins project (see the README of each plugin) and analogous repositories under the 
+`jenkinsci` project on GitHub.  Those URLs are:
+
+* [OpenShift Client Plugin](https://github.com/jenkinsci/openshift-client-plugin)
+* [OpenShift Sync Plugin](https://github.com/jenkinsci/openshift-sync-plugin)
+* [OpenShift Login Plugin](https://github.com/jenkinsci/openshift-login-plugin)
+* [OpenShift Pipeline Plugin (only in maintenance at this point)](https://github.com/jenkinsci/openshift-pipeline-plugin)
+
+As the development process for each of these plugins are common in many ways, we are documenting the process here under 
+the repository for the OpenShift Jenkins images, where each of the plugins will reference this document in their respective 
+READMEs.
+
+## Constructing you development sandbox
+
+So you'll need an environment that has
+
+* Maven (the `mvn` command)
+* Git
+* Java
+* an IDE or editor of your choosing
+
+The Jenkins open source project already has a bunch of useful links on setting up your development environment, including 
+launching Jenkins from the IDE (though there are pros and cons for developing this way vs. developing against a Jenkins 
+server running in an OpenShift pod).
+
+Here are a few of those links for reference:
+
+* https://wiki.jenkins.io/display/JENKINS/Extend+Jenkins (note, there are many useful child pages under this wiki page)  
+* https://wiki.jenkins.io/display/JENKINS/Plugin+tutorial (one of the key child pages under Extend Jenkins)
+* https://wiki.jenkins.io/display/JENKINS/Setting+up+Eclipse+to+build+Jenkins (has concepts common to other IDEs undoubtedly)
+
+### Basic code development flow
+
+Our plugins are constructed such that if they are running in an OpenShift pod, they can determine how to connect to the 
+associated OpenShift master automatically, and no configuration of the plugin from the Jenkins console is needed.
+
+If you choose to run Jenkins outside of an OpenShift pod, then if you want to test interaction with an OpenShift master,
+you'll need to manually configure the plugin so it can connect.  See each plugin's README or the OpenShift documentation
+for the specifics.
+
+An example flow when running in an OpenShift pod then might be:
+
+1. Install maven (platform specific)
+2. Clone this git repository:
+    ```
+    git clone https://github.com/openshift/<plugin in question>-plugin.git
+    ```
+3. In the root of the local repository, run maven 
+    ```
+    cd <plugin in question>-plugin
+    mvn
+    ```
+4. Maven will build target/<plugin name>.hpi  (the Jenkins plugin binary)
+5. Open Jenkins in your browser, and navigate (as an administrator):
```

Comment:
, and log in as an administrator

Reply:
How about `Open Jenkins in your browser, log in as an administrator, and navigate as follows:`

Reply:
That sounds good